neo/rawio/nixrawio.py

```diff
@@ -169,38 +178,47 @@ def _parse_header(self):
         self._generate_minimal_annotations()
         for blk_idx, blk in enumerate(self.file.blocks):
             bl_ann = self.raw_annotations['blocks'][blk_idx]
-            for props in blk.metadata.inherited_properties():
-                self._add_annotate(bl_ann, props, 'blk')
-            for seg_index, seg in enumerate(blk.groups):
-                seg_ann = bl_ann['segments'][seg_index]
-                for props in seg.metadata.inherited_properties():
-                    self._add_annotate(seg_ann, props, 'seg')
-                ansig_idx = 0
-                for da in seg.data_arrays:
-                    if da.type == 'neo.analogsignal' and seg_ann['signals'] != []:
-                        ana_an = seg_ann['signals'][ansig_idx]
-                        sigch_an = self.raw_annotations['signal_channels'][ansig_idx]
-                        for props in da.metadata.inherited_properties():
-                            self._add_annotate(ana_an, props, 'asig')
-                            self._add_annotate(sigch_an, props, 'asig')
-                        ansig_idx += 1
+            props = blk.metadata.inherited_properties()
+            bl_ann.update(self._filter_properties(props, "block"))
+            for grp_idx, grp in enumerate(blk.groups):
+                seg_ann = bl_ann['segments'][grp_idx]
+                props = grp.metadata.inherited_properties()
+                seg_ann.update(self._filter_properties(props, "segment"))
+                sig_idx = 0
+                groupdas = NixIO._group_signals(grp.data_arrays)
+                for nix_name, signals in groupdas.items():
+                    da = signals[0]
+                    if da.type == 'neo.analogsignal' and seg_ann['signals']:
+                        # collect and group DataArrays
+                        sig_ann = seg_ann['signals'][sig_idx]
+                        # sig_chan_ann = self.raw_annotations['signal_channels'][sig_idx]
+                        props = da.metadata.inherited_properties()
+                        sig_ann.update(self._filter_properties(props, 'analogsignal'))
+                        # sig_chan_ann.update(self._filter_properties(props, 'analogsignal'))
```

Comment:
do you stil need these commented code snippets?

Reply:
They're part of the functions that need to be fixed, so it's a little bit in a WIP stage.  If you don't mind the cruft, I'd rather keep them for now.